addons/web/static/src/js/widgets/domain_selector.js

```diff
@@ -471,13 +472,23 @@ var DomainSelector = DomainTree.extend({
      * If the internal domain value was already equal to the given one, this
      * does nothing.
      *
-     * @param {Array|string} domain
+     * @param {string} domain
      * @returns {Deferred} resolved when the rerendering is finished
      */
     setDomain: function (domain) {
-        if (Domain.prototype.arrayToString(domain) === Domain.prototype.arrayToString(this.getDomain())) {
+        if (domain === Domain.prototype.arrayToString(this.getDomain())) {
             return $.when();
         }
+        try {
+            domain = Domain.prototype.stringToArray(domain);
+            this.invalidDomain = false;
+        } catch (err) {
+            // TODO: domain could contain `parent` for example, which is
+            // currently not handled by the DomainSelector
+            this.invalidDomain = true;
+            this.children = [];
```

Comment:
no duplicated code, please :-)

Reply:
Extracted to a function, but you should check the result is OK.